src/appleseed/renderer/kernel/intersection/assemblytree.h

```diff
@@ -32,7 +32,9 @@
 
 // appleseed.renderer headers.
 #include "renderer/kernel/intersection/curvetree.h"
+#ifdef APPLESEED_WITH_EMBREE
```

Comment:
I don't know if it's necessary here. The intent was to hide this check inside the header to keep outer code cleaner.

Reply:
We do the check outside for all our others optional dependencies.
I did this change for consistency.